src/middleware/externalRequests.js

```diff
@@ -63,8 +64,213 @@ const prepareRequestConfig = requestDetails => {
+  let response, error
+
+  if (err.response) {
+    response = err.response
+
+    // Axios response has the data property not the body
+    response.body = response.data
+
+    if (response.status >= 500) {
+      if (request.primary) {
+        setKoaResponseBodyFromPrimary(ctx, request, response.data)
+
+        ctx.primaryReqFailError = true
+        ctx.status = response.status
+      } else {
+        ctx.secondaryFailError = true
+
+        setKoaResponseBody(ctx, request, response.data)
+      }
+    } else {
+      if (request.primary) {
+        setKoaResponseBodyFromPrimary(ctx, request, response.data)
+
+        ctx.primaryCompleted = true
+        ctx.status = response.status
+      } else {
+        ctx.secondaryCompleted = true
+
+        setKoaResponseBody(ctx, request, response.data)
+      }
+    }
+  } else {
+    if (request.primary) {
+      setKoaResponseBodyFromPrimary(ctx, request, err.message)
+
+      ctx.primaryReqFailError = true
+      ctx.status = 500
+    } else {
+      ctx.secondaryFailError = true
+
+      setKoaResponseBody(ctx, request, err.message)
+    }
+    error = {message: err.message}
+  }
+
+  return {response, error}
+}
+
+// Sets the koa response body from the primary request's response body
+const setKoaResponseBodyFromPrimary = (ctx, request, body) => {
+  ctx.hasPrimaryRequest = true
+  ctx.body = {}
+  ctx.body[request.id] = body
+}
+
+// Sets the koa response body if there is no primary request
+const setKoaResponseBody = (ctx, request, body) => {
+  if (!ctx.hasPrimaryRequest) {
+    ctx.body[request.id] = body
+  }
+}
+
+const createAxiosConfig = (request, body) => {
+  const axiosConfig = {
+    url: request.url,
+    method: request.method,
+    body: body
+  }
+
+  if (
+    request.credentials &&
+    request.credentials.username &&
+    request.credentials.password
+  ) {
+    axiosConfig.auth = {
+      username: request.credentials.username,
+      password: request.credentials.password
+    }
+  }
+
+  if (request.headers) {
+    axiosConfig.headers = request.headers
+  }
+
+  return axiosConfig
+}
+
+const sendMappedObject = (ctx, axiosConfig, request, body) => {
+  const reqTimestamp = new Date()
+  let response, error, responseTimestamp
+
+  return axios(axiosConfig)
+    .then(resp => {
+      response = resp
+      response.body = resp.data
+      responseTimestamp = new Date()
+
+      if (request.primary) {
+        setKoaResponseBodyFromPrimary(ctx, request, response.body)
+
+        ctx.status = response.status
+      } else {
+        setKoaResponseBodyFromPrimary(ctx, request, response.body)
+      }
+    })
+    .catch(err => {
+      responseTimestamp = new Date()
+
+      const result = handleRequestError(ctx, request, err)
+      response = result.response
+      error = result.error
+    })
+    .finally(() => {
+      if (ctx.request.header && ctx.request.header['X-OpenHIM-TransactionID']) {
+        const orch = createOrchestration(
```

Comment:
@bradsawadye very minor but please change this variable from `orch` to `orchestration`. This isn't a standard abbreviation for the word so this makes it harder to find in the future.

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/36/commits/33ce80e3923c2410737742821500d48142edcc6f